core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -263,30 +249,12 @@ private static OutputFile newPartitionStatsFile(
   private static PartitionStats recordToPartitionStats(StructLike record) {
     PartitionStats stats =
         new PartitionStats(
-            record.get(PARTITION_POSITION, StructLike.class),
-            record.get(SPEC_ID_POSITION, Integer.class));
-    stats.set(DATA_RECORD_COUNT_POSITION, record.get(DATA_RECORD_COUNT_POSITION, Long.class));
-    stats.set(DATA_FILE_COUNT_POSITION, record.get(DATA_FILE_COUNT_POSITION, Integer.class));
-    stats.set(
-        TOTAL_DATA_FILE_SIZE_IN_BYTES_POSITION,
-        record.get(TOTAL_DATA_FILE_SIZE_IN_BYTES_POSITION, Long.class));
-    stats.set(
-        POSITION_DELETE_RECORD_COUNT_POSITION,
-        record.get(POSITION_DELETE_RECORD_COUNT_POSITION, Long.class));
-    stats.set(
-        POSITION_DELETE_FILE_COUNT_POSITION,
-        record.get(POSITION_DELETE_FILE_COUNT_POSITION, Integer.class));
-    stats.set(
-        EQUALITY_DELETE_RECORD_COUNT_POSITION,
-        record.get(EQUALITY_DELETE_RECORD_COUNT_POSITION, Long.class));
-    stats.set(
-        EQUALITY_DELETE_FILE_COUNT_POSITION,
-        record.get(EQUALITY_DELETE_FILE_COUNT_POSITION, Integer.class));
-    stats.set(TOTAL_RECORD_COUNT_POSITION, record.get(TOTAL_RECORD_COUNT_POSITION, Long.class));
-    stats.set(LAST_UPDATED_AT_POSITION, record.get(LAST_UPDATED_AT_POSITION, Long.class));
-    stats.set(
-        LAST_UPDATED_SNAPSHOT_ID_POSITION,
-        record.get(LAST_UPDATED_SNAPSHOT_ID_POSITION, Long.class));
+            record.get(0, StructLike.class), // partition
```

Comment:
Using raw numbers directly in code is considered bad practice. Static analysis tools flag them as magic numbers and recommend replacing them with named constants. This enhances code readability and makes maintenance easier.

Reply:
done